src/extensions/segment-integration.ts

```diff
@@ -77,7 +77,8 @@ const createSegmentIntegration = (posthog: PostHog): SegmentPlugin => {
             logger.info('No userId set, resetting PostHog')
             posthog.reset()
         }
-        if (ctx.event.userId && ctx.event.userId !== posthog.get_distinct_id()) {
+
+        if (checkIdentity && ctx.event.userId && ctx.event.userId !== posthog.get_distinct_id()) {
```

Comment:
I think the reason this was done this way was so that if the integration was added later on, the linking of distinct id would still work. 

This would now only work if you set up posthog at the same time as segment _or_ if they are sending identify events on page load or whatever. 

Probably fine but it's been a while since I've worked on this so it might be worth validating properly.